Legality/Tables.cs

```diff
@@ -444,6 +444,68 @@ public static partial class Legal
             538, 539, // Sawk & Throh
         };
 
+        #region Legality; unhatchable from egg
+
+        internal static readonly int[] NoHatchFromEgg =
+        {
+            132, // Ditto
+            144, // Articuno
+            145, // Zapdos
+            146, // Moltres
+            150, // Mewtwo
+            151, // Mew
+            201, // Unown
+            243, // Raikou
+            244, // Entei
+            245, // Suicune
+            249, // Lugia
+            250, // Ho-Oh
+            251, // Celebi
+            377, // Regirock
+            378, // Regice
+            379, // Registeel
+            380, // Latias
+            381, // Latios
+            382, // Kyogre
+            383, // Groudon
+            384, // Rayquaza
+            385, // Jirachi
+            386, // Deoxys
+            480, // Uxie
+            481, // Mesprit
+            482, // Azelf
+            483, // Dialga
+            484, // Palkia
+            485, // Heatran
+            486, // Regigigas
+            487, // Giratina
```

Comment:
I went back to my copy of SS to take a look, the level 1 Giratina that I have does not reflect that it hatched from an egg in the summary screen (I can't remember how Gen IV summary works, it should say "hatched" if it came from an egg, right?).

Also, the legality checker does not run for pre-Gen VI Pokemon, so such Pokemon will not be flagged by the legality checker.